src/checker.py

```diff
@@ -67,7 +67,11 @@ def _dump_manifest(self, path):
         dump_manifest(contents, path)
 
     def _collect_external_data(self, path, json_data):
-        for module in json_data.get('modules', []):
+        modules = json_data.get('modules', [])
+        if modules is None:
+            log.error("Failed to deserialize \"modules\" in %s", path)
+            return
+        for module in modules:
```

Comment:
`error` definitely seems too high severity here, and the proposed message is not correct – it was successfully deserialized, as `None`. I personally wouldn't bother logging anything. I suggest:

```python3
modules = json_data.get("modules")
if not modules:
    return
for module in modules:
```

Reply:
Maybe check if `modules` is a `list`? Any other type probably should be ignored.

Reply:
Would this be fine?
```python
        modules = json_data.get('modules')
        if modules is None:
            return
        elif not isinstance(modules, list):
            log.warning("Failed to deserialize \"modules\" in %s", path)
            return
```

Reply:
Sure, I'd prefer something like:

```python3
            log.warning('"modules" in %s is not a list', path)
```

(or with the quotes the other way around, to taste)